File: frontend/src/api/team.ts
import { apiClient } from "./client";

export interface TeamMember {
  id: string;
  hotel_id: string;
  user_id: string;
  role: "owner" | "manager" | "viewer";
  invited_email: string | null;
  created_at: string;
  user_email: string | null;
  user_name: string | null;
}

export const teamApi = {
  list: () => apiClient.get<TeamMember[]>("/api/team"),

  invite: (email: string, role: "manager" | "viewer") =>
    apiClient.post<TeamMember>("/api/team", { email, role }),

  updateRole: (memberId: string, role: "manager" | "viewer") =>
    apiClient.patch<TeamMember>(`/api/team/${memberId}`, { role }),

  remove: (memberId: string) =>
    apiClient.delete(`/api/team/${memberId}`),
};

File: backend/app/schemas/team.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, EmailStr


class TeamMemberOut(BaseModel):
    id: uuid.UUID
    hotel_id: uuid.UUID
    user_id: uuid.UUID
    role: Literal["owner", "manager", "viewer"]
    invited_email: str | None
    created_at: datetime
    user_email: str | None = None
    user_name: str | None = None

    model_config = {"from_attributes": True}


class TeamInviteCreate(BaseModel):
    email: EmailStr
    role: Literal["manager", "viewer"] = "viewer"


class TeamMemberPatch(BaseModel):
    role: Literal["manager", "viewer"]
